app/test/e2e/v2/get-area.spec.js

```diff
@@ -182,32 +197,69 @@ describe('Get areas - V2', () => {
             [{ userId: USERS.USER.id }, { userId: USERS.MANAGER.id }, { userId: USERS.ADMIN.id }]
         );
 
+        // First sync areas
+        const syncResponse = await requester.post(`/api/v2/area/sync`).send({ loggedUser: USERS.ADMIN });
+        syncResponse.status.should.equal(200);
+        syncResponse.body.should.have.property('data').and.be.an('object');
+        syncResponse.body.data.should.have.property('syncedAreas').and.equal(0);
+        syncResponse.body.data.should.have.property('createdAreas').and.equal(3);
+
         // Requesting all areas => should return 5 areas
         const response = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true`);
         response.status.should.equal(200);
         response.body.should.have.property('data').and.be.an('array').and.have.length(5);
 
-        mockSubscriptionFindAll(
-            ['123', '456', '789'],
-            [{ userId: USERS.USER.id }, { userId: USERS.MANAGER.id }, { userId: USERS.ADMIN.id }]
-        );
-
         // Requesting all areas with status saved => should return 1 area
         const savedResponse = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true&status=saved`);
         savedResponse.status.should.equal(200);
         savedResponse.body.should.have.property('data').and.be.an('array').and.have.length(4);
 
-        mockSubscriptionFindAll(
-            ['123', '456', '789'],
-            [{ userId: USERS.USER.id }, { userId: USERS.MANAGER.id }, { userId: USERS.ADMIN.id }]
-        );
-
         // Requesting all areas with status pending => should return 4 areas
         const pendingResponse = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true&status=pending`);
         pendingResponse.status.should.equal(200);
         pendingResponse.body.should.have.property('data').and.be.an('array').and.have.length(1);
     });
 
+    it('Getting areas with all=true filter returns the correct paginated result', async () => {
+        const area1 = await new Area(createArea()).save();
+        const area2 = await new Area(createArea()).save();
+        const area3 = await new Area(createArea()).save();
+        const area4 = await new Area(createArea()).save();
+        const area5 = await new Area(createArea()).save();
+
+        const response = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true&page[size]=3`);
+        response.status.should.equal(200);
+        response.body.should.have.property('data').with.lengthOf(3);
+        response.body.should.have.property('links').and.be.an('object');
+
+        const ids = response.body.data.map((area) => area.id);
+        ids.should.contain(area1.id);
```

Comment:
https://medium.com/building-ibotta/testing-arrays-and-objects-with-chai-js-4b372310fe6d

Reply:
Thanks for the tip, quite nice :ok_hand: